src/generic/gentheme.tcl

```diff
@@ -473,22 +473,27 @@ if {[dict exists $tmpgendict theme scaling ]} {
         }
         rename tk_messageBox _tk_messageBox
         proc tk_messageBox {args} {
-        variable ::ttk::dialog_module::window_name
-	    if [ winfo exists $window_name ] {
-		    raise $window_name
-		    if [ llength $::ttk::dialog_module::args ] {
-			    if [ dict exists $::ttk::dialog_module::args -title ] {
-		    puts "Warning: [ dict get $::ttk::dialog_module::args -title ] dialog is already open, close it first"
-		    	} else {
-		    puts "Warning: a dialog is already open, close it first"
-			}
-		    }
-		    return
-	    } else {
-	    set ::ttk::dialog_module::args $args
-            bell
-            ttk::messageBox {*}$args
-    		}
+            global jobid
+            variable ::ttk::dialog_module::window_name
+            hdbgui eval {INSERT INTO JOBOUTPUT VALUES($jobid, 0, $args)}
+            if [ winfo exists $window_name ] {
+                raise $window_name
+                if [ llength $::ttk::dialog_module::args ] {
+                    if [ dict exists $::ttk::dialog_module::args -title ] {
+                        set message "Warning: [ dict get $::ttk::dialog_module::args -title ] dialog is already open, close it first"
+                        puts $message
```

Comment:
minor: this puts call and the one on line 487 could be combined and placed ahead of the hdbgui call on line 489, since that's a common codepath for all message handling.

Reply:
Updated as advised